libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2499,7 +2500,7 @@ MemberAccess const* SMTEncoder::isEmptyPush(Expression const& _expr) const
 	)
 	{
 		auto const& funType = dynamic_cast<FunctionType const&>(*funCall->expression().annotation().type);
-		if (funType.kind() == FunctionType::Kind::ArrayPush)
+		if (funType.kind() == FunctionType::Kind::ArrayPush || funType.kind() == FunctionType::Kind::ByteArrayPush)
```

Comment:
I think we also need to add this in `arrayPushPopAssign`

Reply:
I have checked it and I don't think that can happen. If I have an expression `e.push()` then `e` cannot be a `ByteArrayPush` because that returns `byte` so another `push` on that is not possible.
`ByteArrayPush` can only be the *innermost* push in a sequence of nested pushes.

Reply:
Ok. Can you then add an assert and a comment related to that at the location I suggested?